### ISSUE TITLE
Make relevant exception for an invalid object inheritance in Entity

### ISSUE THREAD

**Issue by eugene-davis** [IndustrialDragonfly/DEdC#43]
In Entity's getAssociativeArray, the else statement should throw a (possibly custom to the data model) exception if the object being stored doesn't fall into an expected inheritance chain.

**Comment by JoshuaJClark** [comment]
is there any particular reason this cant be a badFunctionCallException?

**Comment by eugene-davis** [comment]
Yes, if we make this a more custom exception, then we could check for it specifically and handle it differently in the calling code based on that fact, whereas if everything is placed into the badFunctionCallException we would have to check the message to know what threw it.
